python/lsst/validate/drp/matchedVisitMetricsTask.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import division, print_function, absolute_import
+
+import os
+
+from lsst.pipe.base import CmdLineTask, ArgumentParser, TaskRunner
+from lsst.pex.config import Config, Field
+from lsst.meas.base.forcedPhotCcd import PerTractCcdDataIdContainer
+from lsst.utils import getPackageDir
+from lsst.validate.base import load_metrics
+from .validate import runOneFilter, plot_metrics
+
+
+class MatchedVisitMetricsRunner(TaskRunner):
+    """Subclass of TaskRunner for MatchedVisitMetrics
+
+    This class transforms the processed
+    arguments generated by the ArgumentParser into the arguments expected by
+    MatchedVisitMetricsTask.run().
+    """
+
+    @staticmethod
+    def getTargetList(parsedCmd, **kwargs):
+        # organize data IDs by filter
+        idListDict = {}
+        for ref in parsedCmd.id.refList:
+            idListDict.setdefault(ref.dataId["filter"], []).append(ref.dataId)
+        # we call run() once with each filter
+        return [(parsedCmd.butler,
+                 filterName,
+                 idListDict[filterName],
+                 ) for filterName in sorted(idListDict.keys())]
+
+    def __call__(self, args):
+        task = self.TaskClass(config=self.config, log=self.log)
+        return task.run(*args)
+
+
+class MatchedVisitMetricsConfig(Config):
+    outputPrefix = Field(
+        dtype=str, default=".",
+        doc="Root path for output files."
+    )
+    metricsFile = Field(
+        dtype=str, optional=True,
+        doc="Full path to metrics file, or None to use metrics in validate_drp."
+    )
+    brightSnr = Field(
+        dtype=float, default=100,
+        doc="Minimum PSF signal-to-noise ratio for a star to be considered bright."
+    )
+    safeSnr = Field(
+        dtype=float, default=50,
+        doc="Minimum median PSF signal-to-noise ratio for a match to be considered safe."
+    )
+    makeJson = Field(
+        dtype=bool, default=True,
+        doc="Whether to write JSON outputs."
+    )
+    makePlots = Field(
+        dtype=bool, default=True,
+        doc="Whether to write plot outputs."
+    )
+    matchRadius = Field(
+        dtype=float, default=1.0,
+        doc="Match radius (arcseconds)."
+    )
+    useJointCal = Field(
+        dtype=bool, default=False,
+        doc="Whether to use jointcal (or meas_mosaic) to calibrate measurements"
+    )
+
+
+class MatchedVisitMetricsTask(CmdLineTask):
+    """An alternate command-line driver for the validate_drp metrics.
+
+    MatchedVisitMetricsTask is very much an incomplete CmdLineTask - it uses
+    the usual metchanisms to define its inputs and read them using a Butler,
+    but writes outputs manually to files with a configuration-defined prefix
+    (config.outputPrefix).  Because the CmdLineTask machinery always creates an
+    output Butler repository, however, it is necessary to run this task with
+    both an output directory and an output prefix, with the former essentially
+    unused.
```

Comment:
Having to specify an unused `--output` seemed annoying.
Can you override this requirement with a default set in the default config for MatchedVisitMetricsTask ?